aospy/data_loader.py

```diff
@@ -0,0 +1,416 @@
+    to intvl_in tags.
+
+    This is the simplest DataLoader; it is useful for instance if one is
+    dealing with raw model history files, which tend to group all variables
+    of a single output interval into single filesets.
+
+    Parameters
+    ----------
+    file_map : dict
+        A dict mapping an input interval to a list of files
+
+    Examples
+    --------
+    Case of two sets of files, one with monthly average output, and one with
+    3-hourly output.
+
+    >>> file_map = {'monthly': '000[4-6]0101.atmos_month.nc',
+    ...             '3hr': '000[4-6]0101.atmos_8xday.nc'}
+    >>> data_loader = DictDataLoader(file_map)
+    """
+    def __init__(self, file_map=None):
+        """Create a new DictDataLoader"""
+        self.file_map = file_map
+
+    def _generate_file_set(self, var=None, start_date=None, end_date=None,
+                           domain=None, intvl_in=None, dtype_in_vert=None,
+                           dtype_in_time=None, intvl_out=None):
+        """Returns the file_set for the given interval in."""
+        try:
+            return self.file_map[intvl_in]
+        except KeyError:
+            raise KeyError('File set does not exist for the specified'
+                           ' intvl_in {0}'.format(intvl_in))
+
+
+class NestedDictDataLoader(DataLoader):
+    """A data loader that locates files based on a nested dictionary mapping of
+    intvl_in to dictionaries mapping variable names to lists of files.
+
+    This is the most flexible existing type of DataLoader; it allows for the
+    specification of different sets of files for different variables.
+
+    Parameters
+    ----------
+    file_map : dict
+        A dict mapping intvl_in to dictionaries mapping Var
+        objects to lists of files
+
+    Examples
+    --------
+    Case of a set of monthly average files for large scale precipitation,
+    and another monthly average set of files for convective precipitation.
+
+    >>> file_map = {'monthly': {'precl': '000[4-6]0101.precl.nc',
+    ...                         'precc': '000[4-6]0101.precc.nc'}}
+    >>> data_loader = NestedDictDataLoader(file_map)
+    """
+    def __init__(self, file_map=None):
+        """Create a new NestedDictDataLoader"""
+        self.file_map = file_map
+
+    def _generate_file_set(self, var=None, start_date=None, end_date=None,
+                           domain=None, intvl_in=None, dtype_in_vert=None,
+                           dtype_in_time=None, intvl_out=None):
+        for name in var.names:
+            try:
+                return self.file_map[intvl_in][name]
+            except KeyError:
+                pass
+        raise KeyError('Files for the var {0} cannot be found in for the '
+                       'intvl_in {1} in this'
+                       ' OneDirDataLoader'.format(var, intvl_in))
+
+
+class GFDLDataLoader(DataLoader):
+    """A data loader that locates files based on GFDL post-processing naming
```

Comment:
One-line summary: "Data Loader for NOAA GFDL model output"

And afterwards define GFDL (=NOAA Geophysical Fluid Dynamics Laboratory)